### PR TITLE
FLUME-3355. Upgrade Kafka to 2.3.1.

### PR DESCRIPTION
**Pull request by iwasakims**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -116,7 +116,7 @@ limitations under the License.
     <xalan.version>2.7.2</xalan.version>
     <xerces.version>2.9.1</xerces.version>
     <wiremock.version>1.53</wiremock.version>
-    <zookeeper.version>3.4.5</zookeeper.version>
+    <zookeeper.version>3.4.13</zookeeper.version>
```
Comment (tmgstevens) [inline pom.xml:119]: Hi, This looks good to me, but is the Zookeeper uplift necessary for this change? Unless ZK 3.4.13 has something specific we need, I'd be minded to separate that into a different JIRA. Much of the community seems to be lagging on Zookeeper and I'd rather not get out of step.
Thanks
Tristan

Comment (iwasakims) [reply pom.xml:119]: @tmgstevens We need to upgrade ZooKeeper to 3.4.7 at least for Kafka 2.1.0 and above which uses functionality backported on [ZOOKEEPER-2237](https://issues.apache.org/jira/browse/ZOOKEEPER-2237). Unit tests fail with following error if ``zookeeper.version`` is 3.4.5, though there is no compilation error.
```
2020-02-16 12:43:05,908 (main) [ERROR - org.slf4j.helpers.MarkerIgnoringBase.error(MarkerIgnoringBase.java:159)] Fatal error during KafkaServer startup. Prepare to shutdown
java.lang.NoClassDefFoundError: org/apache/zookeeper/AsyncCallback$MultiCallback
        at kafka.zk.KafkaZkClient$.apply(KafkaZkClient.scala:1825)
        at kafka.server.KafkaServer.kafka$server$KafkaServer$$createZkClient$1(KafkaServer.scala:363)
        at kafka.server.KafkaServer.initZkClient(KafkaServer.scala:387)
        at kafka.server.KafkaServer.startup(KafkaServer.scala:207)
        at kafka.server.KafkaServerStartable.startup(KafkaServerStartable.scala:38)
        at org.apache.flume.source.kafka.KafkaSourceEmbeddedKafka.<init>(KafkaSourceEmbeddedKafka.java:84)
        at org.apache.flume.source.kafka.TestKafkaSource.startKafkaServer(TestKafkaSource.java:112)
        ...(snip)
```

Comment (tmgstevens) [reply pom.xml:119]: Cool

**Comment by tmgstevens** [comment]
LGTM!
I'll commit this unless anyone objects?

**Comment by tmgstevens** [comment]
Thanks for your contribution @iwasakims

**Comment by iwasakims** [comment]
Thanks, @tmgstevens.

**Pull request merged**